### PR TITLE
Barbarians 1.11.2

### PR DESCRIPTION
**Pull request by Asherslab**
# Changes proposed in this pull request:
- Same as 1.10.2 branch, just for 1.11.2 instead =D

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/EntityAIMeleeGuard.java [hunk 1]
```diff
@@ -124,14 +129,20 @@ protected AIState searchTarget()
      */
     protected AIState huntDown()
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/EntityAIMeleeGuard.java:130]: MAJOR Reduce the number of returns of this method 7, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)

File: src/main/java/com/minecolonies/coremod/proxy/CommonProxy.java [hunk 2]
```diff
@@ -97,11 +107,27 @@ public void registerEntities()
...
         final ResourceLocation locationFishHook = new ResourceLocation(Constants.MOD_ID, "Fishhook");
+        final ResourceLocation locationBarbarian = new ResourceLocation(Constants.MOD_ID, "Barbarian");
+        final ResourceLocation locationArcherBarbarian = new ResourceLocation(Constants.MOD_ID, "ArcherBarbarian");
+        final ResourceLocation locationChiefBarbarian = new ResourceLocation(Constants.MOD_ID, "ChiefBarbarian");
 
         // Half as much tracking range and same update frequency as a player
         // See EntityTracker.addEntityToTracker for more default values
-        EntityRegistry.registerModEntity(locationCitizen, EntityCitizen.class, "Citizen", getNextEntityId(), MineColonies.instance, 256, 2, true);
-        EntityRegistry.registerModEntity(locationFishHook, EntityFishHook.class, "Fishhook", getNextEntityId(), MineColonies.instance, 250, 5, true);
+        EntityRegistry.registerModEntity(locationCitizen, EntityCitizen.class, "Citizen", getNextEntityId(), MineColonies.instance, Constants.ENTITY_TRACKING_RANGE, Constants.ENTITY_UPDATE_FREQUENCY, true);
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/proxy/CommonProxy.java]: MAJOR Split this 206 characters long line (which is greater than 180 authorized). [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00103)

File: src/main/java/com/minecolonies/coremod/proxy/CommonProxy.java [hunk 3]
```diff
@@ -97,11 +107,27 @@ public void registerEntities()
...
+        EntityRegistry.registerModEntity(locationFishHook, EntityFishHook.class, "Fishhook", getNextEntityId(), MineColonies.instance, Constants.ENTITY_TRACKING_RANGE, Constants.ENTITY_UPDATE_FREQUENCY_FISHHOOK, true);
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/proxy/CommonProxy.java]: MAJOR Split this 218 characters long line (which is greater than 180 authorized). [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00103)

File: src/main/java/com/minecolonies/coremod/proxy/CommonProxy.java [hunk 4]
```diff
@@ -97,11 +107,27 @@ public void registerEntities()
...
+        EntityRegistry.registerModEntity(locationBarbarian, EntityBarbarian.class, "Barbarian", getNextEntityId(), MineColonies.instance, Constants.ENTITY_TRACKING_RANGE, Constants.ENTITY_UPDATE_FREQUENCY, true);
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/proxy/CommonProxy.java]: MAJOR Split this 212 characters long line (which is greater than 180 authorized). [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00103)

File: src/main/java/com/minecolonies/coremod/proxy/CommonProxy.java [hunk 5]
```diff
@@ -97,11 +107,27 @@ public void registerEntities()
...
+        EntityRegistry.registerModEntity(locationArcherBarbarian, EntityArcherBarbarian.class, "ArcherBarbarian", getNextEntityId(), MineColonies.instance, Constants.ENTITY_TRACKING_RANGE, Constants.ENTITY_UPDATE_FREQUENCY, true);
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/proxy/CommonProxy.java]: MAJOR Split this 230 characters long line (which is greater than 180 authorized). [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00103)

File: src/main/java/com/minecolonies/coremod/proxy/CommonProxy.java [hunk 6]
```diff
@@ -97,11 +107,27 @@ public void registerEntities()
...
+        EntityRegistry.registerModEntity(locationChiefBarbarian, EntityChiefBarbarian.class, "ChiefBarbarian", getNextEntityId(), MineColonies.instance, Constants.ENTITY_TRACKING_RANGE, Constants.ENTITY_UPDATE_FREQUENCY, true);
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/proxy/CommonProxy.java]: MAJOR Split this 227 characters long line (which is greater than 180 authorized). [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00103)

File: src/main/java/com/minecolonies/coremod/proxy/CommonProxy.java [hunk 7]
```diff
@@ -15,24 +19,30 @@
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.util.ResourceLocation;
 import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.storage.loot.LootTableList;
 import net.minecraftforge.common.MinecraftForge;
 import net.minecraftforge.fml.common.network.NetworkRegistry;
 import net.minecraftforge.fml.common.registry.EntityRegistry;
 import net.minecraftforge.fml.common.registry.GameRegistry;
+import scala.tools.nsc.backend.icode.analysis.CopyPropagation;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/proxy/CommonProxy.java]: MINOR Remove this unused import 'scala.tools.nsc.backend.icode.analysis.CopyPropagation'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/sounds/ModSoundEvents.java [hunk 8]
```diff
@@ -17,6 +17,13 @@
      * List of sound handlers.
      */
     public static final List<AbstractWorkerSounds> handlers = new ArrayList<>();
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/sounds/ModSoundEvents.java]: MINOR Make this member "protected". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2386)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 91 issues
* ![MAJOR][MAJOR] 48 major
* ![MINOR][MINOR] 40 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/4278f5716013c4ef03cef1d77cfee5b420f01865/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.301 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L53](https://github.com/Minecolonies/minecolonies/blob/4278f5716013c4ef03cef1d77cfee5b420f01865/src/main/java/com/minecolonies/coremod/colony/Colony.java#L53): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 46 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L53](https://github.com/Minecolonies/minecolonies/blob/4278f5716013c4ef03cef1d77cfee5b420f01865/src/main/java/com/minecolonies/coremod/colony/Colony.java#L53): class "Colony" has 89 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L212](https://github.com/Minecolonies/minecolonies/blob/4278f5716013c4ef03cef1d77cfee5b420f01865/src/main/java/com/minecolonies/coremod/colony/Colony.java#L212): Assign this magic number 5 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L612](https://github.com/Minecolonies/minecolonies/blob/4278f5716013c4ef03cef1d77cfee5b420f01865/src/main/java/com/minecolonies/coremod/colony/Colony.java#L612): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L1060](https://github.com/Minecolonies/minecolonies/blob/4278f5716013c4ef03cef1d77cfee5b420f01865/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1060): The Cyclomatic Complexity of this method "onWorldTick" is 17 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [Colony.java#L1096](https://github.com/Minecolonies/minecolonies/blob/4278f5716013c4ef03cef1d77cfee5b420f01865/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1096): Assign this magic number 20 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1097](https://github.com/Minecolonies/minecolonies/blob/4278f5716013c4ef03cef1d77cfee5b420f01865/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1097): Assign this magic number 60 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1226](https://github.com/Minecolonies/minecolonies/blob/4278f5716013c4ef03cef1d77cfee5b420f01865/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1226): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1228](https://github.com/Minecolonies/minecolonies/blob/4278f5716013c4ef03cef1d77cfee5b420f01865/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1228): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/com/minecolonies/coremod/sounds/ModSoundEvents.java [hunk 9]
```diff
@@ -15,8 +16,18 @@
 {
     /**
      * List of sound handlers.
+     *
+     * (Making this Protected isn't viable in current state)
      */
+    @SuppressWarnings(Suppression.MAKE_PROTECTED)
     public static final List<AbstractWorkerSounds> handlers = new ArrayList<>();
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/sounds/ModSoundEvents.java:23]: MINOR Make this member "protected". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2386)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 85 issues
* ![MAJOR][MAJOR] 43 major
* ![MINOR][MINOR] 39 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/476a3869f0f71e0fa21a5a877b491d3d55f76296/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.301 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L53](https://github.com/Minecolonies/minecolonies/blob/476a3869f0f71e0fa21a5a877b491d3d55f76296/src/main/java/com/minecolonies/coremod/colony/Colony.java#L53): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 46 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L53](https://github.com/Minecolonies/minecolonies/blob/476a3869f0f71e0fa21a5a877b491d3d55f76296/src/main/java/com/minecolonies/coremod/colony/Colony.java#L53): class "Colony" has 89 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L212](https://github.com/Minecolonies/minecolonies/blob/476a3869f0f71e0fa21a5a877b491d3d55f76296/src/main/java/com/minecolonies/coremod/colony/Colony.java#L212): Assign this magic number 5 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L612](https://github.com/Minecolonies/minecolonies/blob/476a3869f0f71e0fa21a5a877b491d3d55f76296/src/main/java/com/minecolonies/coremod/colony/Colony.java#L612): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L1060](https://github.com/Minecolonies/minecolonies/blob/476a3869f0f71e0fa21a5a877b491d3d55f76296/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1060): The Cyclomatic Complexity of this method "onWorldTick" is 17 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [Colony.java#L1096](https://github.com/Minecolonies/minecolonies/blob/476a3869f0f71e0fa21a5a877b491d3d55f76296/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1096): Assign this magic number 20 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1097](https://github.com/Minecolonies/minecolonies/blob/476a3869f0f71e0fa21a5a877b491d3d55f76296/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1097): Assign this magic number 60 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1226](https://github.com/Minecolonies/minecolonies/blob/476a3869f0f71e0fa21a5a877b491d3d55f76296/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1226): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1228](https://github.com/Minecolonies/minecolonies/blob/476a3869f0f71e0fa21a5a877b491d3d55f76296/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1228): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by marvin-bitterlich** [comment]
LGTM

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/1324/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Comment by Asherslab** [comment]
@Raycoms @OrionDevelopment =D I'd approve my own PR but I don't know that I can, and definitely shouldn't, lol

**Comment by Asherslab** [comment]
Apparently it's auto approved because it's my PR (Auto approved by me) lol

**Comment by Raycoms** [comment]
:shipit: 

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/1324/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Comment by Asherslab** [comment]
Need to wait I've got one small change for better Timing reliability

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 85 issues
* ![MAJOR][MAJOR] 43 major
* ![MINOR][MINOR] 39 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/4c2da1ada79c2428468a1ff3cd1fb03c2570cf03/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.301 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L53](https://github.com/Minecolonies/minecolonies/blob/4c2da1ada79c2428468a1ff3cd1fb03c2570cf03/src/main/java/com/minecolonies/coremod/colony/Colony.java#L53): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 46 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L53](https://github.com/Minecolonies/minecolonies/blob/4c2da1ada79c2428468a1ff3cd1fb03c2570cf03/src/main/java/com/minecolonies/coremod/colony/Colony.java#L53): class "Colony" has 89 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L212](https://github.com/Minecolonies/minecolonies/blob/4c2da1ada79c2428468a1ff3cd1fb03c2570cf03/src/main/java/com/minecolonies/coremod/colony/Colony.java#L212): Assign this magic number 5 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L612](https://github.com/Minecolonies/minecolonies/blob/4c2da1ada79c2428468a1ff3cd1fb03c2570cf03/src/main/java/com/minecolonies/coremod/colony/Colony.java#L612): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L1060](https://github.com/Minecolonies/minecolonies/blob/4c2da1ada79c2428468a1ff3cd1fb03c2570cf03/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1060): The Cyclomatic Complexity of this method "onWorldTick" is 17 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [Colony.java#L1096](https://github.com/Minecolonies/minecolonies/blob/4c2da1ada79c2428468a1ff3cd1fb03c2570cf03/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1096): Assign this magic number 20 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1097](https://github.com/Minecolonies/minecolonies/blob/4c2da1ada79c2428468a1ff3cd1fb03c2570cf03/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1097): Assign this magic number 60 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1226](https://github.com/Minecolonies/minecolonies/blob/4c2da1ada79c2428468a1ff3cd1fb03c2570cf03/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1226): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1228](https://github.com/Minecolonies/minecolonies/blob/4c2da1ada79c2428468a1ff3cd1fb03c2570cf03/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1228): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 85 issues
* ![MAJOR][MAJOR] 43 major
* ![MINOR][MINOR] 39 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/ce31a4daa98e6e8e383a276a4429c203d545032d/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.301 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L53](https://github.com/Minecolonies/minecolonies/blob/ce31a4daa98e6e8e383a276a4429c203d545032d/src/main/java/com/minecolonies/coremod/colony/Colony.java#L53): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 46 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L53](https://github.com/Minecolonies/minecolonies/blob/ce31a4daa98e6e8e383a276a4429c203d545032d/src/main/java/com/minecolonies/coremod/colony/Colony.java#L53): class "Colony" has 89 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L212](https://github.com/Minecolonies/minecolonies/blob/ce31a4daa98e6e8e383a276a4429c203d545032d/src/main/java/com/minecolonies/coremod/colony/Colony.java#L212): Assign this magic number 5 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L612](https://github.com/Minecolonies/minecolonies/blob/ce31a4daa98e6e8e383a276a4429c203d545032d/src/main/java/com/minecolonies/coremod/colony/Colony.java#L612): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L1060](https://github.com/Minecolonies/minecolonies/blob/ce31a4daa98e6e8e383a276a4429c203d545032d/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1060): The Cyclomatic Complexity of this method "onWorldTick" is 17 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [Colony.java#L1096](https://github.com/Minecolonies/minecolonies/blob/ce31a4daa98e6e8e383a276a4429c203d545032d/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1096): Assign this magic number 20 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1097](https://github.com/Minecolonies/minecolonies/blob/ce31a4daa98e6e8e383a276a4429c203d545032d/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1097): Assign this magic number 60 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1226](https://github.com/Minecolonies/minecolonies/blob/ce31a4daa98e6e8e383a276a4429c203d545032d/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1226): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1228](https://github.com/Minecolonies/minecolonies/blob/ce31a4daa98e6e8e383a276a4429c203d545032d/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1228): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**